### PR TITLE
Fixed #5757 and empty application bug

### LINKED ISSUE #5757: [Server] Sample application deployment failure: k8s handler
#### Current Behavior
Using Local Provider, deploying any Application fails with the following error log:

```
DEBU[0251] retrieved session for user with id: meshery
DEBU[0251] session for user with id: meshery was read in tact.
ERRO[2022-07-04T17:24:21-05:00] failed to find k8s handler                    app=meshery code=2012 probable-cause= severity=2 short-description="unable to Create Comp Config..failed to find k8s handler" suggested-remediation=
```

Screen Shot 2022-07-04 at 5 24 17 PM

#### Environment

Meshery has active connection to Kubernetes and Meshery Operator deployed with MeshSync connected.

- **Host OS:** Mac 
- **Platform:** Kubernetes
- **Meshery Server Version:** stable-v06.0.rc-6
- **Meshery Client Version:** stable-v06.0.rc-6

---
#### Contributor [Guides](https://docs.meshery.io/project/contributing) and Resources
- 🛠 [Meshery Build & Release Strategy](https://docs.meshery.io/project/build-and-release)
- 📚 [Instructions for contributing to documentation](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#documentation-contribution-flow)
   - Meshery documentation [site](https://docs.meshery.io/) and [source](https://github.com/meshery/meshery/tree/master/docs)
- 🎨 Wireframes and designs for Meshery UI in [Figma](https://www.figma.com/file/SMP3zxOjZztdOLtgN4dS2W/Meshery-UI)
- 🙋🏾🙋🏼 Questions: [Layer5 Discussion Forum](https://discuss.layer5.io) and [Layer5 Community Slack](http://slack.layer5.io)

### PR DESCRIPTION
**Pull request by Revolyssup**
Signed-off-by: Ashish Tiwari <ashishjaitiwari15112000@gmail.com>

**Description**

This PR fixes #5757 

**Notes for Reviewers**

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Comment by Revolyssup** [comment]
This also fixes https://app.clickup.com/t/2f85une

File: models/pattern/core/pattern.go [hunk 1]
```diff
@@ -341,7 +341,9 @@ ManifestLoop:
 			}
 			return pattern, ErrParseK8sManifest(err)
 		}
-
+		if len(manifest) == 0 {
+			continue
```
Comment (leecalcote) [inline models/pattern/core/pattern.go:345]: Is continuing the right thing to do? @Revolyssup @sudo-NithishKarthik

Comment (leecalcote) [reply models/pattern/core/pattern.go:345]: Is there a response regarding an empty file being set and sent somewhere else?

Comment (Revolyssup) [reply models/pattern/core/pattern.go:345]: When an empty file is sent,this logic won't be executed as that yaml will contain 0 manifests and logic will never enter this loop

**Comment by leecalcote** [comment]
Lint failures...

File: models/pattern/core/pattern.go (hunk 1 above)
Comment (leecalcote) [reply models/pattern/core/pattern.go:358]: Makes sense. Thanks for this.

File: models/pattern/core/pattern.go [hunk 2]
```diff
@@ -323,14 +323,27 @@ func processCytoElementsWithPattern(eles []cytoscapejs.Element, pf *Pattern, cal
 	}
 	return nil
 }
+func manifestIsEmpty(manifests []string) bool {
+	for _, m := range manifests {
+		x := strings.TrimSpace(strings.Trim(m, "\n"))
+		if x != "---" && x != "" {
```
Comment (leecalcote) [inline models/pattern/core/pattern.go:329]: Won't this match on long manifests that have multiple `---` separators?

File: models/pattern/core/pattern.go [hunk 3]
```diff
@@ -323,14 +323,27 @@ func processCytoElementsWithPattern(eles []cytoscapejs.Element, pf *Pattern, cal
...
 		Name:     "Autogenerated",
 		Services: map[string]*Service{},
 	}
 
 	manifests := strings.Split(data, "\n---\n")
-
+	//For `---` separated manifests, even if only one manifest is there followed/preceeded by multiple `\n---\n`- the manifest be will be valid
+	//If there is no data present (except \n---\n) , then the yaml will be marked as empty and error will be thrown
+	if manifestIsEmpty(manifests) {
+		return pattern, ErrParseK8sManifest(fmt.Errorf("No data in the passed manifest"))
```
Comment (leecalcote) [inline models/pattern/core/pattern.go]:
```suggestion
		return pattern, ErrParseK8sManifest(fmt.Errorf("No data in application manifest"))
```

**Review by leecalcote — CHANGES_REQUESTED** [review]

File: models/pattern/core/pattern.go (hunk 2 above)
Comment (Revolyssup) [reply models/pattern/core/pattern.go:329]: This logic essentially says - After separating by "\n---\n", if we find some other characters except for "---" - we mark that as a non-empty manifest

File: models/pattern/core/pattern.go [hunk 4]
```diff
@@ -323,14 +323,27 @@ func processCytoElementsWithPattern(eles []cytoscapejs.Element, pf *Pattern, cal
 	}
 	return nil
 }
+func manifestIsEmpty(manifests []string) bool {
```
Comment (leecalcote) [inline models/pattern/core/pattern.go:326]:
```suggestion
// This logic essentially says: After separating by "\n---\n", if we find some other characters except for "---" - we mark that as a non-empty manifest
func manifestIsEmpty(manifests []string) bool {
```

File: models/pattern/core/pattern.go [hunk 5]
```diff
@@ -462,12 +478,15 @@ func createPatternServiceFromExtendedK8s(manifest map[string]interface{}) (strin
...
 	}
+
 	id := name
 	uid, err := uuid.NewV4()
 	if err == nil {
 		id = uid.String()
 	}
-
+	if apiVersion == "" || kind == "" {
+		return "", Service{}, ErrCreatePatternService(fmt.Errorf("empty apiVersion or kind"))
```
Comment (leecalcote) [inline models/pattern/core/pattern.go:488]:
```suggestion
		return "", Service{}, ErrCreatePatternService(fmt.Errorf("empty apiVersion or kind in manifest"))
```

**Pull request merged**